pkg/bundle/internal/truststore/types_test.go

```diff
@@ -23,12 +23,65 @@ import (
 	"testing"
 
 	"github.com/pavlo-v-chernykh/keystore-go/v4"
+	"github.com/stretchr/testify/assert"
 
 	"github.com/cert-manager/trust-manager/pkg/apis/trust/v1alpha1"
 	"github.com/cert-manager/trust-manager/pkg/util"
 	"github.com/cert-manager/trust-manager/test/dummy"
 )
 
+func Test_Encoder_Deterministic(t *testing.T) {
+	tests := map[string]struct {
+		encoder             Encoder
+		expNonDeterministic bool
+	}{
+		"JKS default password": {
+			encoder: NewJKSEncoder(v1alpha1.DefaultJKSPassword),
+		},
+		"JKS custom password": {
+			encoder: NewJKSEncoder("my-password"),
+		},
+		"PKCS default password": {
+			encoder: NewPKCS12Encoder(v1alpha1.DefaultPKCS12Password),
+		},
+		"PKCS custom password": {
+			encoder: NewPKCS12Encoder("my-password"),
+			// FIXME: We should try to make all encoders deterministic
+			expNonDeterministic: true,
+		},
+	}
+
+	for name, test := range tests {
+		// test := test // NOTE: uncomment for Go < 1.22, see /doc/faq#closures_and_goroutines
```

Comment:
suggestion: I'd be happy to remove this commented line. Go versions below 1.22 aren't supported by Go any more, so I don't think we need to help people using them!

Reply:
But we require Go > 1.22, ref. https://github.com/cert-manager/trust-manager/blob/main/go.mod#L3

Reply:
I might be missing something here 😅 

Go 1.22 is the [version](https://tip.golang.org/doc/go1.22) which removed the shadowing issue, right? We require go 1.22+ as you say and go 1.21 is out of support now.

This comment is only relevant for go 1.21 and below, isn't it? So it can be removed?

Am I missing something?

Reply:
Ugh, sorry! 😊 I added this copy-pasted comment. It's removed now. Appologies!